### PR TITLE
ISPN-11780 updating xsite docs for IRAC implementation

### PR DESCRIPTION
**Pull request by oraNod**
https://issues.redhat.com/browse/ISPN-11780

### CODE REVIEW

**Comment by tristantarrant** [comment]
Shouldn't we mention the vector clocks too?

**Comment by oraNod** [comment]
> Shouldn't we mention the vector clocks too?

Can do. I wasn't sure if saying vector clocks specifically was too low level and refer to them more abstractly as "metadata" for keys. I seem to recall that being feedback from a previous review but am happy to go into more detail.

**Comment by tristantarrant** [comment]
The problem here is that, by just saying we are comparing site names lexicographically to resolve conflicts, it seems like our implementation is trivial, which is not the case.

File: documentation/src/main/asciidoc/topics/con_xsite_concurrent_writes.adoc [hunk 1]
```diff
@@ -1,59 +1,51 @@
...
-//ordering wins.
+However if a client puts "k1=5" in **LON** at the same time that a client puts
+"k1=8" in **NYC**, {brandname} detects a conflicting entry because the vector
+value for "k1" is strictly greater or less between **LON** and **NYC**.
 
-//Following the same example, if "k1" conflicts between **LON** and **NYC**,
-//{brandname} always uses "k1" in **LON**.
+When it finds conflicting entries, {brandname} uses the Java `compareTo(String
+anotherString)` method to compare site names. To determine which key takes
+priority, {brandname} selects the site name that is lexicographically greater
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/con_xsite_concurrent_writes.adoc:44]: I may be confused, but I think it is lexicographically lower/lesser... A is preceding B.

File: documentation/src/main/asciidoc/titles/xsite/xsite.asciidoc [hunk 2]
```diff
@@ -14,7 +14,12 @@ include::../{topics}/attributes/community-attributes.adoc[]
 :stem: asciimath
 
 //Title
-= Replicating Data Across Sites with {brandname} {infinispanversion}
+= {brandname} {infinispanversion} Guide to Cross-Site Replication
```
[the remark this answers is not in the document]
Comment (oraNod) [reply documentation/src/main/asciidoc/titles/xsite/xsite.asciidoc:17]: sounds good

File: documentation/src/main/asciidoc/topics/con_xsite_concurrent_writes.adoc [hunk 3]
```diff
@@ -1,59 +1,51 @@
...
-//backup. If {brandname} finds entries that conflict, it compares site names
-//based on alphabetical order. The site name with the highest alphabetical
-//ordering wins.
+However if a client puts "k1=5" in **LON** at the same time that a client puts
+"k1=8" in **NYC**, {brandname} detects a conflicting entry because the vector
+value for "k1" is strictly greater or less between **LON** and **NYC**.
 
-//Following the same example, if "k1" conflicts between **LON** and **NYC**,
-//{brandname} always uses "k1" in **LON**.
+When it finds conflicting entries, {brandname} uses the Java `compareTo(String
```
[the remark this answers is not in the document]
Comment (oraNod) [reply documentation/src/main/asciidoc/topics/con_xsite_concurrent_writes.adoc:42]: added. thanks.

File: documentation/src/main/asciidoc/topics/con_xsite_concurrent_writes.adoc [hunk 4]
```diff
@@ -1,59 +1,51 @@
...
+{brandname} replicates to **NYC**. A client then puts "k1=3" in **NYC** and the
+vector clock updates to `1,1`, which {brandname} replicates to **LON**.
 
-//{brandname} uses version information for keys to detect conflicts during a
-//backup. If {brandname} finds entries that conflict, it compares site names
-//based on alphabetical order. The site name with the highest alphabetical
-//ordering wins.
+However if a client puts "k1=5" in **LON** at the same time that a client puts
+"k1=8" in **NYC**, {brandname} detects a conflicting entry because the vector
+value for "k1" is strictly greater or less between **LON** and **NYC**.
```
[the remark this answers is not in the document]
Comment (oraNod) [reply documentation/src/main/asciidoc/topics/con_xsite_concurrent_writes.adoc:40]: @pruivo I'm not sure. if the vector value is greater or less then it means there is a conflict, right? 

maybe we should say if the vector value for "k1" does not match?

[the remark this answers is not in the document]
Comment (pruivo) [reply documentation/src/main/asciidoc/topics/con_xsite_concurrent_writes.adoc:40]: This is my point of view:
`[1,1]` is "strictly" less than `[2,1]`, `[1,2]` (no conflict)
`[2,1]` is "strictly" greater than `[1,1]`, `[0,1]`, `[1,0]` (no conflict)
`[1,2]` is conflicting with `[2,1]`

so, I have no idea how to put that in words xD

[the remark this answers is not in the document]
Comment (oraNod) [reply documentation/src/main/asciidoc/topics/con_xsite_concurrent_writes.adoc]: thanks @pruivo that's a great catch.

File: documentation/src/main/asciidoc/topics/con_xsite_concurrent_writes.adoc [hunk 5]
```diff
@@ -1,59 +1,61 @@
...
+Following the same example, to resolve the conflict for "k1", {brandname} uses
+the value for "k1" that originates from **LON**. This results in "k1=5" in both
+**LON** and **NYC** after {brandname} resolves the conflict and replicates the
+value.
+
+[TIP]
+====
+Prepend site names with numbers as a simple way to represent the order of
+priority for resolving conflicting entries; for example, **1LON** and **2NYC**.
+====
```
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/con_xsite_concurrent_writes.adoc:57]: @tristantarrant How about this as a tip to assign priority to sites?

**Comment by pruivo** [comment]
integrated! thanks @oraNod !

**Pull request merged**